OpenRA.Game/Map/Map.cs

```diff
@@ -65,137 +25,26 @@ public enum MapVisibility
-		readonly Type type;
-
-		readonly string key;
-		readonly string fieldName;
-		readonly bool required;
-		readonly string ignoreIfValue;
-
-		public MapField(string key, string fieldName = null, bool required = true, string ignoreIfValue = null)
-		{
-			this.key = key;
-			this.fieldName = fieldName ?? key;
-			this.required = required;
-			this.ignoreIfValue = ignoreIfValue;
-
-			field = typeof(Map).GetField(this.fieldName);
-			property = typeof(Map).GetProperty(this.fieldName);
-			if (field == null && property == null)
-				throw new InvalidOperationException("Map does not have a field/property " + fieldName);
-
-			var t = field != null ? field.FieldType : property.PropertyType;
-			type = t == typeof(List<MiniYamlNode>) ? Type.NodeList :
-				t == typeof(MiniYaml) ? Type.MiniYaml : Type.Normal;
-		}
-
-		public void Deserialize(Map map, List<MiniYamlNode> nodes)
-		{
-			var node = nodes.FirstOrDefault(n => n.Key == key);
-			if (node == null)
-			{
-				if (required)
-					throw new YamlException($"Required field `{key}` not found in map.yaml");
-				return;
-			}
-
-			if (field != null)
-			{
-				if (type == Type.NodeList)
-					field.SetValue(map, node.Value.Nodes);
-				else if (type == Type.MiniYaml)
-					field.SetValue(map, node.Value);
-				else
-					FieldLoader.LoadField(map, fieldName, node.Value.Value);
-			}
-
-			if (property != null)
-			{
-				if (type == Type.NodeList)
-					property.SetValue(map, node.Value.Nodes, null);
-				else if (type == Type.MiniYaml)
-					property.SetValue(map, node.Value, null);
-				else
-					FieldLoader.LoadField(map, fieldName, node.Value.Value);
-			}
-		}
-
-		public void Serialize(Map map, List<MiniYamlNode> nodes)
-		{
-			var value = field != null ? field.GetValue(map) : property.GetValue(map, null);
-			if (type == Type.NodeList)
-			{
-				var listValue = (List<MiniYamlNode>)value;
-				if (required || listValue.Count > 0)
-					nodes.Add(new MiniYamlNode(key, null, listValue));
-			}
-			else if (type == Type.MiniYaml)
-			{
-				var yamlValue = (MiniYaml)value;
-				if (required || (yamlValue != null && (yamlValue.Value != null || yamlValue.Nodes.Count > 0)))
-					nodes.Add(new MiniYamlNode(key, yamlValue));
-			}
-			else
-			{
-				var formattedValue = FieldSaver.FormatValue(value);
-				if (required || formattedValue != ignoreIfValue)
-					nodes.Add(new MiniYamlNode(key, formattedValue));
-			}
-		}
+		Map Load(ModData modData, IReadOnlyPackage package);
+		Map Create(ModData modData, ITerrainInfo terrainInfo, int width, int height);
+		string ComputeUID(ModData modData, IReadOnlyPackage package);
+		void UpdatePreview(ModData modData, MapPreview mapPreview, IReadOnlyPackage p, IReadOnlyPackage parent, MapClassification classification, string[] mapCompatibility, MapGridType gridType);
 	}
 
-	public class Map : IReadOnlyFileSystem
+	public abstract class Map : IReadOnlyFileSystem
 	{
-		public const int SupportedMapFormat = 11;
-		public const int CurrentMapFormat = 12;
```

Comment:
This is never added back

Reply:
Once again not sure what is meant here, `SupportedMapFormat` exists and `MapFormat` is used for the map's format.

Reply:
See #19891

Reply:
Have added this back now.